pyccel/ast/datatypes.py

```diff
@@ -488,6 +488,16 @@ class HomogeneousContainerType(ContainerType):
     This is the case for objects such as arrays, lists, etc.
     """
     __slots__ = ()
+    _name = None
+
+    @property
+    def name(self):
+        """
+        Get the name of the container.
+        
+        Get the name of the container.
+        """
+        return self._name
```

Comment:
If I remember correctly, it was because of this line (line 1197 in `ccode.py`) :
```python
container_type = 'hset_' if dtype.name == 'set' else 'vec_'
```
we were accessing the private attribute directly (`dtype._name`) so I created a getter property for it.

Reply:
I understand we can just check on the class type without needing this

Reply:
Ah ok. That makes sense :+1: I just wanted to check it wasn't being misused
Please can you move this getter to `PyccelType` so all types can use this

Reply:
updated d0c218b344206f673c1a4977a25b9b16d806d227